src/runtime/methodbinder.cs

```diff
@@ -455,6 +467,18 @@ internal Binding Bind(IntPtr inst, IntPtr args, IntPtr kw)
             }
             return null;
         }
+        /// <summary>
+        /// Returns true if src type (int -type) is promotable to a higher resolution type (float|double)
+        /// </summary>
+        /// <param name="src"></param>
+        /// <param name="dst"></param>
+        /// <returns></returns>
+        private bool SimplePromotableType(Type src, Type dst)
+        {
+            return (((src == typeof(int) || src == typeof(short)) && (dst == typeof(double) || dst == typeof(float))))
+                || ((src == typeof(long)) && (dst == typeof(double)))
```

Comment:
Great, then we could try the approach I suggested in my initial comment, - just use the standard conversion function inside pythonnet. That's wider approach, but there is a hope it's consistently used for argument conversions, and thus we can terminate this discussion, since then the arguments to a constructor undergoes (almost) the same rules as an ordinary pythonnet function.
I am on vacation now, but can try to see if its possible to get in touch with a computer long enough to produce the change.